kbase-extension/static/kbase/js/widgets/appWidgets2/input/dynamicDropdownInput.js

```diff
@@ -122,24 +127,33 @@ define([
          * Sets the dropdown value to the given value. Constructs an id from it that (should)
          * be unique enough to apply to the dropdown.
          */
-        function setControlValue(value) {
+        function updateControlValue() {
+            const value = model.value;
+            const displayValue = model.displayValue || {};
             const control = ui.getElement('input-container.input');
-            if ($(control).find('option[value="' + value + '"]').length) {
-                $(control).val(value).trigger('change');
-            } else {
-                const newOption = new Option(value, value, true, true);
-                $(control).append(newOption).trigger('change');
+            if (model.value) {
+                const selectorValue = value.replace(/"/g, '\\"').replace(/'/g, "\\'");
```

Comment:
This is from an issue with the FTP staging version of the dropdown - if there are quotes in the value (say, a file name with quotes in it), it goes all kinds of wonky. Need to be escaped.